File: leetcode/py/container_with_most_water.py
class Solution:
    def maxArea(self, height):
        _max_area = 0
        left, right = 0, len(height)
        
        while left < right:
            area = (right - left) * min(height[left], height[right])
            _max_area = max(_max_area, area)
            
            if height[left] < height[right]:
                left += 1
                
            else: 
                right -= 1
            
        return _max_area

File: leetcode/js/containerWithMostWater.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left = 0
    let right = height.length -1 
    let maxArea = 0

    while (left<right){
        let area = (right-left) * (Math.min(height[left], height[right]))
        maxArea = Math.max(maxArea,area)
        if(height[left] > height[right]){
            right--
        }
        else{
            left++
        }

    }
    return maxArea
};

var maxArea = function(height) {
    let maxArea = 0;
    let left = 0;
    let right = height.length - 1;

    while (left < right) {
        maxArea = Math.max(maxArea, (right - left) * Math.min(height[left], height[right]));

        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }

    return maxArea;    
};